libfreerdp/codec/test/TestFreeRDPRegion.c

```diff
@@ -644,6 +644,9 @@ static int test_norbert_case() {
 			goto out;
 	}
 
+	if (!compareRectangles(region16_extents(&region), &expected_inter_extents, 1) )
```

Comment:
@hardening 
Regarding your question of 'a test that would demonstrate the need for the changes in region16_intersect_rect()', I reuse the above 'test_norbert_case' to demonstrate the need for the fix in region16_intersect_rect. Without the fix, the result would be 
{0, 0, 1920, 1078}. The original incorrect 'expected_inter_extents' exactly shows the bug